### PR TITLE
399 robot is stuck in infinite loop using translate by with wait=true

### PR DESCRIPTION
**Pull request by FabienDanieau**
The bug was due to the `if wait` that was not supposed to be triggered with unreachable comment (i.e. there is nothing to wait for). It is fixed by a `elif`

```
    response = self._goto_stub.GoToCartesian(request)
    if response.id == -1:
        self._logger.error(f"Target pose:\\n {target} \\nwas not reachable. No command sent.")
    if wait:
        self._logger.info(f"Waiting for movement with {response}.")
        while not self._is_goto_finished(response):
            time.sleep(0.1)
        self._logger.info(f"Movement with {response} finished.")
    return response
```

I've added this mechanism to all goto functions, even if the goto_joints don't raise unreachable errors for now. 
Also there is a timeout in the wait function to be sure that we don't wait forever if something goes bad on the server side (timeout is set to duration + 1 sec.)
\+ little bit of refactoring

### CODE REVIEW

File: src/reachy2_sdk/parts/goto_based_part.py [hunk 1]
```diff
@@ -80,3 +83,14 @@ def _is_goto_finished(self, id: GoToId) -> bool:
...
         )
         return result
+
+    def _wait_goto(self, id: GoToId, timeout: float = 10) -> None:
+        """Wait for a goto to finish. timeout is in seconds."""
+        self._logger_goto.info(f"Waiting for movement with {id}.")
+        t1 = time.time()
+        while not self._is_goto_finished(id):
+            time.sleep(0.1)
+            if time.time() - t1 > timeout:
```
Comment (glannuzel) [inline src/reachy2_sdk/parts/goto_based_part.py:93]: I think timeout for the gotos should be checked once the movement has begun only.

If we do:
```
pose2 = get_pose_matrix([0.4, -0.2, -0.1], [0, -90, 0])
reachy.goto_posture("default")
reachy.r_arm.goto_from_matrix(pose2, wait=True)
```
Then the timeout is triggered, whereas the goto was just queued

**Review by glannuzel — CHANGES_REQUESTED** [review]
I think we should change how timeouts are triggered

File: src/reachy2_sdk/parts/goto_based_part.py (hunk 1 above)
Comment (FabienDanieau) [reply src/reachy2_sdk/parts/goto_based_part.py]: The timeout is now the sum of the duration of all pending movements on the part

**Review by glannuzel — APPROVED** [review]
Awesome!

**Pull request merged**